datafusion/core/src/config.rs

```diff
@@ -237,6 +237,15 @@ impl BuiltInConfigs {
                  to reduce the number of rows decoded.",
                 false,
             ),
+            #[cfg(test)]
+            ConfigDefinition::new_bool(
+                OPT_OPTIMIZER_SKIP_FAILED_RULES,
+                "When set to true, the logical plan optimizer will produce warning \
+                messages if any optimization rules produce errors and then proceed to the next \
+                rule. When set to false, any rules that produce errors will cause the query to fail.",
```

Comment:
```suggestion
                "OVERRIDDEN FOR TEST",
```

Reply:
I think this could be very confusing to people (that tests and defaults are different) so I suggest some more comments to make it clearer